plugins/sub_plugins/validate/jsonschema.py

```diff
@@ -153,30 +220,45 @@ def _validate_jsonschema(self):
         error_messages = None
 
         draft = self._get_sub_plugin_options("draft")
+        check_format = self._get_sub_plugin_options("check_format")
         error_messages = []
 
         for criteria in self._criteria:
-            if draft == "draft3":
-                validator = jsonschema.Draft3Validator(
-                    criteria,
-                    format_checker=jsonschema.draft3_format_checker,
-                )
-            elif draft == "draft4":
-                validator = jsonschema.Draft4Validator(
-                    criteria,
-                    format_checker=jsonschema.draft4_format_checker,
-                )
-            elif draft == "draft6":
-                validator = jsonschema.Draft6Validator(
-                    criteria,
-                    format_checker=jsonschema.draft6_format_checker,
-                )
-            else:
-                validator = jsonschema.Draft7Validator(
-                    criteria,
-                    format_checker=jsonschema.draft7_format_checker,
-                )
-
+            format_checker = None
+            validator_class = None
+            if draft is not None:
+                try:
+                    validator_class = self._JSONSCHEMA_DRAFTS[draft]["validator"]
+                except KeyError:
+                    display.warning(
+                        'No jsonschema validator available for "{draft}", falling back to autodetection. A newer version of jsonschema might support this draft.'.format(
```

Comment:
sanity checks are failing on this line, you need to break it up so it's under 160 characters long.

Reply:
I shortened the line. I find it hard to find the right arguments for ansible-test to test this locally but a simple `ansible-test sanity --docker` did not show any errors that have to do with line length.